inst/EpiModel 2.0 Introduction.rmd

```diff
@@ -0,0 +1,254 @@
+
+# EpiModel Changes
+
+In this section, we will detail the core functions that remain in EpiModel 2.0, but that received an overhaul for efficiency and end-user accesiblility. All changes listed in this section will be further demonstrated in the "EpiModel 1.0: Code Migration" section of this document.
+
+## Mode to Group
+
+The most significant change to the EpiModel workflow is in how we handle heterogenous sub-populations. In the original EpiModel worfklow, designation of a second population under interest, was done by initializing a new network and setting it's ```bipartite``` input parameter equal to the number of actors in the first mode:
+
+```{r echo = TRUE, eval = FALSE}
+num1 <- num2 <- 50
+nw <- network.initialize(num1+num2)
+```
+
+This created a mutli-modal (or bipartite) network in which there were 50 actors in mode one and 50 actors in mode two and in which intermode mixing does not occur (i.e., no dyads match on mode assignment). In order to make modeling in EpiModel more flexible, in EpiModel 2.0 we move away from the language of modes and instead refer to heterogenous sub-populations as groups. This change requires a change in how one sets up initial networks as well:
+
+```{r echo = TRUE, eval = FALSE}
+num1 <- num2 <- 50
+nw <- network.initialize(n = num1+num2)
+nw <- set.vertex.attribute(nw, "group", rep(1:2, c(num1, num2)))
+```
+
+Above we have the updated syntax for creating bipartite, now called "multigroup", networks within the EpiModel workflow. The network is initialized as usual using ```network.initialize```, however, in order to capture the previous two-mode structure, we set heterogeneity as a vertex attribute instead of a network feature; in the case of the new EpiModel workflow, this is done using the attribute ```group```. This change allows for more flexibility both in interpretation as well as network structure: in a ```bipartite``` network, mixing between the two groups is not allowed. However, in a ```multigroup``` network, as group assignment is now a vertex attribute we can model within group mixing. For example, in a population of males and females, this new workflow allows us to model more than just heterosexual relationships as below:
+
+```{r echo = TRUE, eval = FALSE}
+num1 <- num2 <- 50
+nw <- network.initialize(n = num1+num2)
+nw <- set.vertex.attribute(nw, "group", rep(1:2, c(num1, num2)))
+formation <- ~ edges + nodefactor("group")+nodematch("group")
+target.stats <- c(30, 30, 2)
+```
+
+In this network we have a total of 30 relationships in the population, two of which involve individuals of the same sex. We can replicate the traditional bipartite behavior of default network setups by setting the ```nodematch``` formation target statistic to 0:
```

Comment:
Just call them edges consistently, and define what that might mean at the top of the document (e.g., relationships, partnerships, persistent contacts...)

Reply:
Fixed in latest commit.